libarchive/filter_fork_posix.c

```diff
@@ -65,11 +65,25 @@
 #  include <unistd.h>
 #endif
 
+#if defined(HAVE_POSIX_SPAWNP) && defined(__FreeBSD__)
+extern char *const environ[];
```

Comment:
Trying to figure out the best approach here:  What if we added a try-compile configure check to see if `extern char *const environ[]` worked?  Then we could do this whenever `HAVE_ENVIRON_VAR && HAVE_POSIX_SPAWNP`.  That should get us the right behavior on any platform that supported `posix_spawnp` in this fashion.

Reply:
That would probably be a more appropriate approach. I'll take a deeper look into it here soon.

I should probably also research if macOS exposes an `environ` global symbol, or how one would go about fetching the environment array.